scio-jdbc/src/main/scala/com/spotify/scio/jdbc/sharded/ShardQuery.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.jdbc.sharded
+
+sealed trait ShardQuery extends Serializable
+
+case class RangeShardQuery[T](range: Range[T], upperBoundInclusive: Boolean) extends ShardQuery
+case class PrefixShardQuery[T](prefix: T) extends ShardQuery
+
+object ShardQuery {
+
+  private val RANGE_QUERY_TEMPLATE = "SELECT * FROM %s WHERE %s >= %s and %s %s %s"
+  private val PREFIX_QUERY_TEMPLATE = "SELECT * FROM %s WHERE %s LIKE '%s%%'"
+
+  def toSelectStatement(shardQuery: ShardQuery)(tableName: String, shardColumn: String): String =
```

Comment:
I don't think this needs to be curried.

Reply:
yeah, I had some idea, but it isn't needed now.